src/pretix/presale/views/widget.py

```diff
@@ -109,19 +109,24 @@ def widget_css_etag(request, **kwargs):
         return f'{_get_source_cache_key()}-{request.organizer.cache.get_or_set("css_version", default=lambda: int(time.time()))}'
 
 
-def widget_js_etag(request, lang, **kwargs):
+def widget_js_etag(request, version, lang, **kwargs):
     gs = GlobalSettingsObject()
-    return gs.settings.get('widget_checksum_{}'.format(lang))
+    return gs.settings.get('widget_checksum_{}_{}'.format(version, lang))
 
 
 @gzip_page
 @condition(etag_func=widget_css_etag)
 @cache_page(60)
-def widget_css(request, **kwargs):
+def widget_css(request, version, **kwargs):
     o = getattr(request, 'event', request.organizer)
 
+    # TODO: handle version deprecation
+    css_path = 'pretixpresale/scss/widget.v{}.scss'.format(version)
+    if not finders.find(css_path):
```

Comment:
I think if the system just knows which version is the current one, we can avoid this code path. We'll hit it *a lot*.